### PR TITLE
inference: use uuid v7 for most of table

### PR DESCRIPTION
**Pull request by notmd**
use this lib https://github.com/stevesimmons/uuid7

### CODE REVIEW

File: inference/server/oasst_inference_server/user_chat_repository.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ class Config:
     async def get_chats(self) -> list[models.DbChat]:
         query = sqlmodel.select(models.DbChat)
         query = query.where(models.DbChat.user_id == self.user_id)
-        query = query.order_by(models.DbChat.created_at.desc())
+        query = query.order_by(models.DbChat.id.desc())
```
Comment (AbdBarho) [inline inference/server/oasst_inference_server/user_chat_repository.py:20]: could we leave this sorted by date? this has the advantage of being functional regardless of which id type we use. Also perhaps slightly faster but not that much.

File: inference/server/oasst_inference_server/models/worker.py [hunk 2]
```diff
@@ -17,7 +18,7 @@ class WorkerEventType(str, enum.Enum):
 class DbWorkerComplianceCheck(SQLModel, table=True):
     __tablename__ = "worker_compliance_check"
 
-    id: str = Field(default_factory=lambda: str(uuid4()), primary_key=True)
+    id: str = Field(default_factory=lambda: uuid7str(), primary_key=True)
```
Comment (andreaskoepf) [inline inference/server/oasst_inference_server/models/worker.py:21]: is the lambda necessary or could it be `default_factory=uuid7str` ?

Comment (notmd) [reply inference/server/oasst_inference_server/models/worker.py]: Thanks, I'm not really familiar with python syntax.

File: inference/server/oasst_inference_server/user_chat_repository.py (hunk 1 above)
Comment (notmd) [reply inference/server/oasst_inference_server/user_chat_repository.py]: Done!

**Review by olliestanley — APPROVED** [review]

**Pull request merged**